helix-core/src/main/java/org/apache/helix/controller/rebalancer/CustomRebalancer.java

```diff
@@ -132,10 +134,14 @@ private Map<String, String> computeCustomizedBestStateForPartition(
       boolean notInErrorState = currentStateMap != null
           && !HelixDefinedState.ERROR.toString().equals(currentStateMap.get(instance));
       boolean enabled = !disabledInstancesForPartition.contains(instance) && isResourceEnabled;
-
+      InstanceConfig instanceConfig = cache.getInstanceConfigMap().get(instance);
+      boolean hasEvacuatedOp = instanceConfig != null &&
+          instanceConfig.getInstanceOperation().getOperation() == InstanceConstants.InstanceOperation.EVACUATE;
+      boolean isAssignableForCustomizedResource = cache.getLiveInstances().containsKey(instance) && hasEvacuatedOp;
```

Comment:
Can we leave custom rebalancer away with this logic currently? I think this requires a comprehensive design with custom rebalance users in open source.